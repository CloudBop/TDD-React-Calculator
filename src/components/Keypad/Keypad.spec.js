import React from 'react';
import { shallow } from 'enzyme';
import Keypad from './Keypad'

describe('Keypad Unit Tests', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(
    <Keypad
      // mock the expecting props
      callOperator={jest.fn()}
      numbers={[]}
      operators={[]}
      setOperator={jest.fn()}
      updateDisplay={jest.fn()}
    />
  ));

  it('Should compare against snapshot <Keypad/>', () => expect(wrapper).toMatchSnapshot())
  
  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('renders the values of numbers', () => {
    wrapper.setProps({numbers: ['0', '1', '2']});
    expect(wrapper.find('.numbers-container').text()).toEqual('012');
  });
});