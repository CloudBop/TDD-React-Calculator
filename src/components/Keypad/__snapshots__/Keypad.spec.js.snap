// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad Unit Tests Should compare against snapshot <Keypad/> 1`] = `ShallowWrapper {}`;
